apps/examples/src/ai-sdk-example/index.ts

```diff
@@ -0,0 +1,101 @@
+import { env } from "node:process";
+import { AiSdkLlm, LlmAgent, InMemoryRunner, FunctionTool } from "@iqai/adk";
+import { google } from "@ai-sdk/google";
+
+function getWeather(location: string): string {
+	console.log(`🌤️ Getting weather for ${location}...`);
+	const conditions = [
+		{ temp: 22, condition: "sunny", humidity: 65, wind: 5 },
+		{ temp: 15, condition: "cloudy", humidity: 80, wind: 12 },
+		{ temp: 8, condition: "rainy", humidity: 90, wind: 18 },
+	];
+	const weather = conditions[Math.floor(Math.random() * conditions.length)];
+	return `The weather in ${location} is ${weather.temp}°C and ${weather.condition}. Humidity: ${weather.humidity}%, Wind: ${weather.wind} km/h`;
+}
+
+async function demonstrateWeatherAgent() {
+	console.log("🌤️ Weather Agent Demo - Using AI SDK (User brings provider)\n");
+
+	if (!env.GOOGLE_GENERATIVE_AI_API_KEY) {
+		console.error("❌ GOOGLE_GENERATIVE_AI_API_KEY required for this example");
+		console.log(
+			"💡 Users can use any provider they want - just change the import!",
+		);
+		process.exit(1);
+	}
+
+	try {
+		const model = new AiSdkLlm(google("gemini-2.0-flash"));
```

Comment:
instead of letting user wrap `LanguageModel` in `AiSdkLlm()`, we can just directly take the LanguageModel as model input, then model can be `string` | `LanguageModel`. we know to pass LanguageModel to AiSdkLlm so we can handle this logic inside the runner itself.